options/internal/include/bits/types.h

```diff
@@ -1,319 +0,0 @@
-#define __MLIBC_UINT_FAST32_MIN __MLIBC_UINT64_MIN
-
-typedef __mlibc_uint64 __mlibc_uint_fast64;
-#define __MLIBC_UINT_FAST64_C(x) __MLIBC_UINT64_C(x)
-#define __MLIBC_UINT_FAST64_MAX __MLIBC_UINT64_MAX
-#define __MLIBC_UINT_FAST64_MIN __MLIBC_UINT64_MIN
-
-#else
-#  error "Missing architecture specific code"
-#endif
-
-// Special types.
-
-typedef __INTMAX_TYPE__  __mlibc_intmax;
-typedef __INTPTR_TYPE__  __mlibc_intptr;
-typedef __PTRDIFF_TYPE__ __mlibc_ptrdiff;
-#define __MLIBC_INTMAX_MAX __INTMAX_MAX__
-#define __MLIBC_INTMAX_MIN (-__INTMAX_MAX__ - 1)
-#define __MLIBC_INTPTR_MAX __INTPTR_MAX__
-#define __MLIBC_INTPTR_MIN (-__INTPTR_MAX__ - 1)
-#define __MLIBC_PTRDIFF_MAX __PTRDIFF_MAX__
-#define __MLIBC_PTRDIFF_MIN (-__PTRDIFF_MAX__ - 1)
-
-typedef __UINTMAX_TYPE__ __mlibc_uintmax;
-typedef __UINTPTR_TYPE__ __mlibc_uintptr;
-typedef __SIZE_TYPE__    __mlibc_size;
-#define __MLIBC_UINTMAX_MAX __UINTMAX_MAX__
-#define __MLIBC_UINTPTR_MAX __UINTPTR_MAX__
-#define __MLIBC_SIZE_MAX __SIZE_MAX__
-
-// Other limits.
-
-#define __MLIBC_WCHAR_MAX __WCHAR_MAX__
-#define __MLIBC_WCHAR_MIN __WCHAR_MIN__
-
-#define __MLIBC_WINT_MAX __WINT_MAX__
-#define __MLIBC_WINT_MIN __WINT_MIN__
-
-#define __MLIBC_SIG_ATOMIC_MAX __SIG_ATOMIC_MAX__
-#define __MLIBC_SIG_ATOMIC_MIN __SIG_ATOMIC_MIN__
-
-// ----------------------------------------------------------------------------
-// Sanity checking. Make sure that we agree with the compiler's ABI.
-// ----------------------------------------------------------------------------
-
-#if defined(__cpp_static_assert)
-#	define __MLIBC_STATIC_ASSERT(c, text) static_assert(c, text)
-#elif !defined(__cplusplus)
-#	define __MLIBC_STATIC_ASSERT(c, text) _Static_assert(c, text)
-#else
-#	define __MLIBC_STATIC_ASSERT(c, text)
-#endif
-
-#define __MLIBC_CHECK_TYPE(T1, T2) __MLIBC_STATIC_ASSERT(sizeof(T1) == sizeof(T2),\
-	#T1 " != " #T2);
-
-// Least-width.
-__MLIBC_CHECK_TYPE(__mlibc_int8,  __INT_LEAST8_TYPE__);
-__MLIBC_CHECK_TYPE(__mlibc_int16, __INT_LEAST16_TYPE__);
-__MLIBC_CHECK_TYPE(__mlibc_int32, __INT_LEAST32_TYPE__);
-__MLIBC_CHECK_TYPE(__mlibc_int64, __INT_LEAST64_TYPE__);
-
-__MLIBC_CHECK_TYPE(__mlibc_uint8,  __UINT_LEAST8_TYPE__);
-__MLIBC_CHECK_TYPE(__mlibc_uint16, __UINT_LEAST16_TYPE__);
-__MLIBC_CHECK_TYPE(__mlibc_uint32, __UINT_LEAST32_TYPE__);
-__MLIBC_CHECK_TYPE(__mlibc_uint64, __UINT_LEAST64_TYPE__);
-
-// Fast-width.
-// Unfortunately, GCC and Clang disagree about fast types.
-#ifndef __clang__
-	__MLIBC_CHECK_TYPE(__mlibc_int_fast8,  __INT_FAST8_TYPE__);
-	__MLIBC_CHECK_TYPE(__mlibc_int_fast16, __INT_FAST16_TYPE__);
-	__MLIBC_CHECK_TYPE(__mlibc_int_fast32, __INT_FAST32_TYPE__);
-	__MLIBC_CHECK_TYPE(__mlibc_int_fast64, __INT_FAST64_TYPE__);
-
-	__MLIBC_CHECK_TYPE(__mlibc_uint_fast8,  __UINT_FAST8_TYPE__);
-	__MLIBC_CHECK_TYPE(__mlibc_uint_fast16, __UINT_FAST16_TYPE__);
-	__MLIBC_CHECK_TYPE(__mlibc_uint_fast32, __UINT_FAST32_TYPE__);
-	__MLIBC_CHECK_TYPE(__mlibc_uint_fast64, __UINT_FAST64_TYPE__);
-#endif
```

Comment:
Has this been fixed in LLVM? If these asserts still fail, we can't apply the commit. Otherwise, we get an inconsistent ABI and can't link clang-compiled code together with GCC-compiled code.

Reply:
This has been sorted out by the new updates, thank you for pointing it out.